src/graph.ts

```diff
@@ -1,30 +1,88 @@
-import { PackageInfos } from "./types/PackageInfo";
-import { getInternalDeps } from "./dependencies";
+import type { PackageInfos } from "./types/PackageInfo";
+import type { DependencyMap } from "./createDependencyMap";
+import type { PackageGraph, PackageGraphVisitor } from "./types/PackageGraph";
 
-const graphCache = new Map<PackageInfos, [string, string][]>();
+import { createDependencyMap } from "./createDependencyMap";
+import multimatch from "multimatch";
 
-export function getPackageGraph(packages: PackageInfos) {
-  if (graphCache.has(packages)) {
-    return graphCache.get(packages)!;
-  }
+// Reference: https://github.com/pnpm/pnpm/blob/597047fc056dd25b83638a9ab3df0df1c555ee49/packages/filter-workspace-packages/src/parsePackageSelector.ts
+export interface PackageGraphScope {
+  namePatterns?: string[];
+  includeDependencies?: boolean;
+  includeDependents?: boolean;
+}
+
+export function createPackageGraph(packages: PackageInfos, scope: PackageGraphScope = {}): PackageGraph {
+  const dependencyMap = createDependencyMap(packages);
```

Comment:
This map has all dependencies (dev and standard), including external packages. (annotating with my reading of the BL :))